src/Analyzers/CSharp/Analyzers/UseObjectInitializer/CSharpUseNamedMemberInitializerAnalyzer.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.UseCollectionInitializer;
+using Microsoft.CodeAnalysis.UseObjectInitializer;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseObjectInitializer;
+
+internal sealed class CSharpUseNamedMemberInitializerAnalyzer :
+    AbstractUseNamedMemberInitializerAnalyzer<
+        ExpressionSyntax,
+        StatementSyntax,
+        BaseObjectCreationExpressionSyntax,
+        MemberAccessExpressionSyntax,
+        ExpressionStatementSyntax,
+        LocalDeclarationStatementSyntax,
+        VariableDeclaratorSyntax,
+        CSharpUseNamedMemberInitializerAnalyzer>
```

Comment:
previously UseNamedMemberInitializerAnalyzer was a concrete type that worked on both C# and VB.  but it was handling this case wrong, and each language has specific logic it needs.  So this moved to become an  abstract type, and there is now a C# and VB specific subclass.